doc/managing_pachyderm/backup_restore_and_migrate.md

```diff
@@ -0,0 +1,410 @@
+# Backup, Restore, and Migrate
+
+## Introduction
+
```

Comment:
Considering how long this document is, we should at a table of context with some basic anchor links to below sections. I think even having a depth-2 ToC would be good:
- Backup procedures
    - 1...
    - 2...
Migration Procedures
    - 1...